PROTOCOL.md

```diff
@@ -1348,6 +1348,8 @@ When enabled:
 
 Delta supports defining default expressions for columns on Delta tables. Delta will generate default values for columns when users do not explicitly provide values for them when writing to such tables, or when the user explicitly specifies the `DEFAULT` SQL keyword for any such column.
 
+Note that this metadata only applies for write operations, not read operations. For example, SQL DML operations such as INSERT, UPDATE, and MERGE will use the default values, but ALTER TABLE ... ADD COLUMN commands may not specify default values for the new columns at the time of creation (it is acceptable to specify default values for new columns in a later ALTER TABLE ... ALTER COLUMN command, however). Similarly, when reading data from a Delta table, the default values will not be returned for columns that are missing from the data files.
```

Comment:
`at the time of creation`? creation of the column? of the table?

Can you give me concrete example where you CAN and another example where you CANNOT use the `ALTER TABLE ... ADD COLUMN` command? Sorry, it isn't clear to me from this text

Reply:
I think an example like the one here would help: https://issues.apache.org/jira/browse/SPARK-38334.

In my understanding, for Delta it would work like this:

```
CREATE TABLE T(a INT, b INT NOT NULL);

-- The default default is NULL
INSERT INTO T VALUES (DEFAULT, 0);
INSERT INTO T(b)  VALUES (1);
SELECT * FROM T;
(NULL, 0)
(NULL, 1)

-- Adding a default to a table with rows, DO NOT set the values for the
-- existing rows, only to the new rows.
ALTER TABLE T ADD COLUMN c INT DEFAULT 5; -- This is not valid

ALTER TABLE T ADD COLUMN c INT
-- ALTER TABLE TO SET NEW DEFAULT TO 5

INSERT INTO T VALUES (1, 2, DEFAULT);
SELECT * FROM T;
(NULL, 0, NULL) -- This is different from Parquet Tables (NULL, 0, 5)
(NULL, 1, NULL) -- This is different from Parquet Tables (NULL, 1, 5)
(1, 2, 5) 
```

Reply:
Thanks @scottsand-db and @felipepessoto, I updated the protocol according to these suggestions. Please take another look :)